### PR TITLE
Add logging to PVF and other related parts

### PR DESCRIPTION
**Pull request by pepyakin**
This should make it easier to diagnose issues like #3562 , #3581

### CODE REVIEW

File: node/core/pvf/src/execute/queue.rs [hunk 1]
```diff
@@ -175,24 +180,35 @@ fn handle_to_queue(queue: &mut Queue, to_queue: ToQueue) {
...
+			handle_job_finish(queue, worker, outcome, artifact_id, result_tx);
 		},
 	}
 }
 
+fn handle_worker_spawned(queue: &mut Queue, idle: IdleWorker, handle: WorkerHandle) {
+	queue.workers.spawn_inflight -= 1;
+	let worker = queue.workers.running.insert(WorkerData { idle: Some(idle), handle });
+
+	tracing::debug!(target: LOG_TARGET, "worker spawned {:?}", worker);
```
Comment (bkchr) [inline node/core/pvf/src/execute/queue.rs:196]:
```suggestion
	tracing::debug!(target: LOG_TARGET, ?worker, "worker spawned");
```

File: node/core/pvf/src/execute/queue.rs [hunk 2]
```diff
@@ -212,6 +228,14 @@ fn handle_job_finish(queue: &mut Queue, worker: Worker, outcome: Outcome, result
 			(None, Err(ValidationError::InvalidCandidate(InvalidCandidate::AmbigiousWorkerDeath))),
 	};
 
+	tracing::debug!(
+		target: LOG_TARGET,
+		validation_code_hash = ?artifact_id.code_hash,
+		worker_rip = idle_worker.is_none(),
+		"job finished. result: {:?}",
+		result,
```
Comment (bkchr) [inline node/core/pvf/src/execute/queue.rs:236]:
```suggestion
		?result,
		"job finished.",
```

File: node/core/pvf/src/execute/queue.rs [hunk 3]
```diff
@@ -271,15 +297,22 @@ async fn spawn_worker_task(program_path: PathBuf, spawn_timeout: Duration) -> Qu
 ///
 /// The worker must be running and idle.
 fn assign(queue: &mut Queue, worker: Worker, job: ExecuteJob) {
+	tracing::debug!(
+		target: LOG_TARGET,
+		validation_code_hash = ?job.artifact.id,
+		"assigning the execute worker {:?}",
+		worker,
```
Comment (bkchr) [inline node/core/pvf/src/execute/queue.rs:304]:
```suggestion
		?worker,
		"assigning the execute worker",
```

File: node/core/pvf/src/execute/worker.rs [hunk 4]
```diff
@@ -68,30 +68,55 @@ pub enum Outcome {
...
 	);
 
-	if send_request(&mut stream, &artifact_path, &validation_params).await.is_err() {
+	if let Err(err) = send_request(&mut stream, &artifact.path, &validation_params).await {
+		tracing::warn!(
+			target: LOG_TARGET,
+			worker_pid = %pid,
+			validation_code_hash = ?artifact.id.code_hash,
+			"failed to send an execute request: {:?}",
+			err,
```
Comment (bkchr) [inline node/core/pvf/src/execute/worker.rs:90]:
```suggestion
			?error,
			"failed to send an execute request",
```

File: node/core/pvf/src/execute/worker.rs [hunk 5]
```diff
@@ -68,30 +68,55 @@ pub enum Outcome {
...
 		target: LOG_TARGET,
 		worker_pid = %pid,
+		validation_code_hash = ?artifact.id.code_hash,
 		"starting execute for {}",
-		artifact_path.display(),
+		artifact.path.display(),
 	);
 
-	if send_request(&mut stream, &artifact_path, &validation_params).await.is_err() {
+	if let Err(err) = send_request(&mut stream, &artifact.path, &validation_params).await {
```
Comment (bkchr) [inline node/core/pvf/src/execute/worker.rs:84]:
```suggestion
	if let Err(error) = send_request(&mut stream, &artifact.path, &validation_params).await {
```

File: node/core/pvf/src/execute/worker.rs [hunk 6]
```diff
@@ -68,30 +68,55 @@ pub enum Outcome {
...
 			match response {
-				Err(_err) => return Outcome::IoErr,
+				Err(err) => {
+					tracing::warn!(
+						target: LOG_TARGET,
+						worker_pid = %pid,
+						validation_code_hash = ?artifact.id.code_hash,
+						"failed to recv an execute response: {:?}",
+						err,
+					);
```
Comment (bkchr) [inline node/core/pvf/src/execute/worker.rs:105]:
```suggestion
				Err(error) => {
					tracing::warn!(
						target: LOG_TARGET,
						worker_pid = %pid,
						validation_code_hash = ?artifact.id.code_hash,
						?error,
						"failed to recv an execute response",
					);
```

File: node/core/pvf/src/host.rs [hunk 7]
```diff
@@ -550,7 +563,13 @@ async fn sweeper_task(mut sweeper_rx: mpsc::Receiver<PathBuf>) {
 		match sweeper_rx.next().await {
 			None => break,
 			Some(condemned) => {
-				let _ = async_std::fs::remove_file(condemned).await;
+				let result = async_std::fs::remove_file(&condemned).await;
+				tracing::trace!(
+					target: LOG_TARGET,
+					"Sweeping the artifact file {}: {:?}",
+					condemned.display(),
+					result,
```
Comment (bkchr) [inline node/core/pvf/src/host.rs:571]:
```suggestion
					?result,
					"Sweeping the artifact file {}",
					condemned.display(),
```

File: node/core/pvf/src/prepare/pool.rs [hunk 8]
```diff
@@ -224,6 +225,7 @@ fn handle_to_pool(
 			}
 		},
 		ToPool::Kill(worker) => {
+			tracing::debug!(target: LOG_TARGET, "killing the worker {:?}", worker);
```
Comment (bkchr) [inline node/core/pvf/src/prepare/pool.rs:228]:
```suggestion
			tracing::debug!(target: LOG_TARGET, ?worker, "killing worker");
```

File: node/core/pvf/src/prepare/queue.rs [hunk 9]
```diff
@@ -349,6 +362,14 @@ async fn handle_worker_concluded(
 
 	queue.artifact_id_to_job.remove(&artifact_id);
 
+	tracing::debug!(
+		target: LOG_TARGET,
+		validation_code_hash = ?artifact_id.code_hash,
+		"prepare worker {:?} concluded, rip={:?}",
+		worker,
```
Comment (bkchr) [inline node/core/pvf/src/prepare/queue.rs:369]:
```suggestion
		?worker,
		?rip,
		"prepare worker concluded",
```

File: node/core/pvf/src/prepare/queue.rs [hunk 10]
```diff
@@ -349,6 +362,14 @@ async fn handle_worker_concluded(
...
+		rip,
```
Comment (bkchr) [inline node/core/pvf/src/prepare/queue.rs:370]:
```suggestion
```

File: node/core/pvf/src/prepare/queue.rs [hunk 11]
```diff
@@ -380,8 +401,9 @@ async fn handle_worker_concluded(
 }
 
 async fn handle_worker_rip(queue: &mut Queue, worker: Worker) -> Result<(), Fatal> {
-	let worker_data = queue.workers.remove(worker);
+	tracing::debug!(target: LOG_TARGET, "prepare worker {:?} ripped", worker);
```
Comment (bkchr) [inline node/core/pvf/src/prepare/queue.rs:404]:
```suggestion
	tracing::debug!(target: LOG_TARGET, ?worker, "prepare worker ripped");
```

**Review by bkchr — APPROVED** [review]

File: node/core/pvf/src/execute/queue.rs (hunk 1 above)
Comment (pepyakin) [reply node/core/pvf/src/execute/queue.rs]: Just fwiw my predicate on whether to make it a field or part of the formatted string was wether I would personally look up on this value or not. I guess it doesn't really make much sense

**Pull request merged**